src/stores/disk.rs

```diff
@@ -0,0 +1,712 @@
+use crate::IOCached;
+use directories::BaseDirs;
+use instant::Duration;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use sled::Db;
+use std::{fmt::Display, path::PathBuf, time::SystemTime};
+use std::marker::PhantomData;
+
+pub struct DiskCacheBuilder<K, V> {
+    seconds: u64,
+    refresh: bool,
+    namespace: String,
+    prefix: String,
+    disk_path: Option<PathBuf>,
+    _phantom_k: PhantomData<K>,
+    _phantom_v: PhantomData<V>,
+}
+
+const ENV_KEY: &str = "CACHED_DISK_PATH";
+const DEFAULT_NAMESPACE: &str = "cached-disk-store:";
+const LAST_CLEANUP_KEY: &str = "last-cleanup";
+
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum DiskCacheBuildError {
+    #[error("Storage connection error")]
+    ConnectionError(#[from] sled::Error),
+    #[error("Connection string not specified or invalid in env var {env_key:?}: {error:?}")]
+    MissingDiskPath {
+        env_key: String,
+        error: std::env::VarError,
+    },
+}
+
+impl<K, V> DiskCacheBuilder<K, V>
+where
+    K: Display,
+    V: Serialize + DeserializeOwned,
+{
+    /// Initialize a `DiskCacheBuilder`
+    pub fn new<S: AsRef<str>>(prefix: S, seconds: u64) -> DiskCacheBuilder<K, V> {
+        Self {
+            seconds,
+            refresh: false,
+            namespace: DEFAULT_NAMESPACE.to_string(),
+            prefix: prefix.as_ref().to_string(),
+            disk_path: None,
+            _phantom_k: Default::default(),
+            _phantom_v: Default::default(),
+        }
+    }
```

Comment:
I was thinking that we create an individual sled-file for each disk-cache to simplify things. Then the `namespace` and `prefix` could go away and the disk-path would be required. We could keep `fn disk_path` below and repurpose it to generate a valid path for you from a string, so when the macro is constructing a DiskCache it can call `disk_path($function_name)` to get a per-function disk-cache. Thoughts?